apps/shell/tash_init.c

```diff
@@ -160,6 +165,20 @@ static int tash_nonftdi_usbconinit(void)
 int tash_init(void)
 {
 	int ret = OK;
+#ifdef CONFIG_PM
+	pm_domain_arg_t pm_domain_arg;
+	int pmdrv_fd = open(PM_DRIVER, O_WRONLY);
+	pm_domain_arg.domain_name = TASH_PM_DOMAIN_NAME;
+	if (pmdrv_fd >= 0) {
+		if (ioctl(pmdrv_fd, PMIOC_DOMAIN_REGISTER, &pm_domain_arg) != OK) {
+			shdbg("pm_domain_register failed(%d)\n", get_errno());
+		}
+		tash_pm_domain_id = pm_domain_arg.domain_id;
+		close(pmdrv_fd);
+	} else {
+		shdbg("open %s failed(%d), \n", PM_DRIVER, get_errno());
+	}
+#endif
```

Comment:
How about making a new file like tash_pm.c and using tash_pm function in tash main files instead of coding them at several location?

Reply:
Done